File: backend/src/main/java/com/fitness/fitnessaicoach/repository/DailyLogRepository.java
package com.fitness.fitnessaicoach.repository;

import com.fitness.fitnessaicoach.domain.DailyLog;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDate;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface DailyLogRepository extends JpaRepository<DailyLog, UUID> {

    List<DailyLog> findByUserId(UUID userId);

<<<<<<< HEAD
    Optional<DailyLog> findTopByUserIdOrderByLogDateDescIdDesc(UUID userId);

=======
>>>>>>> main
    Optional<DailyLog> findByUserIdAndLogDate(UUID userId, LocalDate date);
}

File: android/fitness-ai-coach-app/app/src/main/java/com/fitness/fitnessaicoach/domain/repository/DailyLogRepository.kt
package com.fitness.fitnessaicoach.domain.repository

import com.fitness.fitnessaicoach.core.result.AppResult
import com.fitness.fitnessaicoach.domain.model.DailyLog

interface DailyLogRepository {
    suspend fun getTodayDailyLog(): AppResult<DailyLog>
    suspend fun saveDailyLog(dailyLog: DailyLog): DailyLog
}
